deep_quoridor/src/agents/alphazero/alphazero.py

```diff
@@ -218,34 +218,50 @@ def train_network(self):
         if len(self.recent_losses) > 100:  # Keep only recent losses
             self.recent_losses = self.recent_losses[-100:]
 
+    def _log_action(
+        self,
+        visit_probs,
+        root_children,
+    ):
+        if not self.action_log.is_enabled():
+            return
+
+        self.action_log.clear()
+
+        _, top_indices = torch.topk(torch.tensor(visit_probs), min(5, len(visit_probs)))
+
+        scores = {root_children[i].action_taken: visit_probs[i] for i in top_indices}
+        self.action_log.action_score_ranking(scores)
+
     def get_action(self, observation) -> int:
         game, _, _ = construct_game_from_observation(observation["observation"])
 
         # Run MCTS to get action visit counts
         root_children = self.mcts.search(game)
         visit_counts = np.array([child.visit_count for child in root_children])
         visit_counts_sum = np.sum(visit_counts)
-
         if visit_counts_sum == 0:
             raise RuntimeError("No nodes visited during MCTS")
-        else:
```

Comment:
I removed the else since it's raising an exception anyway, so it's just a change in indentation